brokerapi/brokers/gcp_service_broker.go

```diff
@@ -433,6 +406,37 @@ func (gcpBroker *GCPServiceBroker) LastOperation(instanceID string) (models.Last
 
 }
 
+func (gcpBroker *GCPServiceBroker) lastOperationAsync(instanceId, serviceId string) (models.LastOperation, error) {
+	done, err := gcpBroker.ServiceBrokerMap[serviceId].PollInstance(instanceId)
+	if err != nil {
```

Comment:
looking good. If you want to go further, I recommend trying out initializing a models.LastOperation at the top and an empty err (perhaps named different?). Set the value of LastOperation.State in the branch as soon as you have determined the state.